### PR TITLE
Qr code navigation prevent

### PR DESCRIPTION
**Pull request by RikuRuokonen**
Added prompt to prevent accidental nagivation away from QR-page. After prompt confirmation added listener that automatically logs user out, if new route is not "checkIn".

### CODE REVIEW

File: admin-frontend/src/components/checkIn.js [hunk 1]
```diff
@@ -21,12 +24,20 @@ const Container = styled.div`
   justify-content: center;
   flex-direction: column;
 `;
-let audio = new Audio(ding)
+
+let audio = new Audio(ding);
```
Comment (ohel) [inline admin-frontend/src/components/checkIn.js]: Any reason this is a let and not a const?

File: admin-frontend/src/components/checkIn.js [hunk 2]
```diff
@@ -88,8 +104,14 @@ const CheckInView = (props) => {
     showNotification('Jokin meni pieleen! Kokeile uudestaan.', 'warning')
   };
 
+  const notRedirectingBack = (location) => !isSubstring(location.pathname, "checkIn");
```
Comment (ohel) [inline admin-frontend/src/components/checkIn.js]: Maybe a bit misleading name as it's checking against a constant path. So, "navigatingToCheckIn" or something?

File: admin-frontend/src/utils.js [hunk 3]
```diff
@@ -35,3 +35,4 @@ export const ageValidator = (value, allValues) => {
 export const getYouthClubs = () => dataProvider(GET_LIST, 'youthClub')
   .then((res) => res.data.map((youthClub) => ({ id: youthClub.name, name: youthClub.name }))) // TODO: Eventually it will be wise to change the ID to the actual ID. This will require backend changes.)
 
+export const isSubstring = (mainString, subString) => mainString.indexOf(subString) !== -1;
```
Comment (ohel) [inline admin-frontend/src/utils.js]: Are we using ES6 i.e. not supporting IE? If ES6, then simply mainstring.contains("substring") is the way to go.

Comment (RikuRuokonen) [reply admin-frontend/src/utils.js]: I assume you mean mainstring.includes("string") ?

File: admin-frontend/src/utils.js [hunk 4]
```diff
@@ -35,3 +35,4 @@ export const ageValidator = (value, allValues) => {
 export const getYouthClubs = () => dataProvider(GET_LIST, 'youthClub')
   .then((res) => res.data.map((youthClub) => ({ id: youthClub.name, name: youthClub.name }))) // TODO: Eventually it will be wise to change the ID to the actual ID. This will require backend changes.)
 
+export const isSubstring = (mainString, subString) => mainString.includes(subString);
```
Comment (ohel) [inline admin-frontend/src/utils.js]: With my previous comment I meant that there's no need for this isSubstring function, just use the string.includes("substring") directly wherever you need it as it's actually a couple of characters shorter even + standard. But fix it in another commit

**Pull request merged**